autoscheduler/scraper/tests/api_tests.py

```diff
@@ -0,0 +1,253 @@
+        expected = {'title': 'Sports Law', 'credit_hours': None}
+
+        # Act
+        serializer = CourseSerializer(self.courses[4])
+
+        # Assert
+        self.assertEqual(expected, serializer.data)
+
+    def test_api_course_gives_valid_response_csce(self):
+        """ Tests that /api/course?dept=CSCE&course_num=181&term=201931 gives the
+            correct output
+        """
+        # Arrange
+        expected = CourseSerializer(self.courses[0])
+        data = {'dept': 'CSCE', 'course_num': '181', 'term': '201931'}
+
+        # Act
+        response = self.client.get("/api/course", data=data)
+
+        # Assert
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.json(), expected.data)
+
+    def test_api_course_gives_valid_response_law(self):
+        """ Tests that /api/course?dept=LAW&course_num=7500S&term=202031 gives the
+            correct output (verifies API can handle null credit_hours)
+        """
+        # Arrange
+        expected = CourseSerializer(self.courses[4])
+        data = {'dept': 'LAW', 'course_num': '7500S', 'term': '202031'}
+
+        # Act
+        response = self.client.get("/api/course", data=data)
+
+        # Assert
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.json(), expected.data)
+
+    def test_api_section_serializer_gives_expected_output(self):
+        """ Tests that the section serializer yields the correct data """
+        # Arrange
+        first_start = time(11, 30)
+        first_end = time(12, 20)
+        second_start = time(9, 10)
+        second_end = time(10)
+        meeting_days = [True] * 7
+        expected = {
+            'crn': 12345,
+            'instructor_gpa': None,
+            'instructor_name': 'Akash Tyagi',
+            'honors_only': False,
+            'meetings': {
+                '10': {
+                    'days': meeting_days,
+                    'start_time': first_start,
+                    'end_time': first_end,
+                    'type': 'LEC',
+                },
+                '11': {
+                    'days': meeting_days,
+                    'start_time': second_start,
+                    'end_time': second_end,
+                    'type': 'LEC',
+                },
+            },
+            'section_num': 501,
+            'web_only': False,
+        }
+
+        # Act
+        serializer = SectionSerializer(self.sections[0])
+
+        # Assert
+        self.assertEqual(expected, serializer.data)
+
+    def test_api_sections_gives_valid_response(self):
+        """ Tests that /api/sections?&dept=CSCE&course_num=310term=201931 gives the
+            correct output
+        """
+        # Arrange
+        first_start = time(11, 30)
+        first_end = time(12, 20)
+        second_start = time(9, 10)
+        second_end = time(10)
+        meeting_days_true = [True] * 7
+        meeting_days_false = [False] * 7
+        expected = {
+            '1': {
```

Comment:
The keys are supposed to be the ids of each section (keys for `meetings` are also their ids) since they're currently being used by frontend in the schedule/meeting cards

Reply:
The keys are supposed to be the ids of each section (keys for `meetings` are also their ids) since they're currently being used by frontend in the schedule/meeting cards

Reply:
ok